### ISSUE TITLE
License

### ISSUE THREAD

**Issue by pboutros** [uclahs-cds/public-R-BoutrosLab-plotting-general#81]
Add UCLA copyright information to all files

**Comment by aholmes** [comment]
@pboutros are you referring to this part of the GPLv2?

```
<one line to give the program's name and a brief idea of what it does.>
Copyright (C) <year>  <name of author>

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License along
with this program; if not, write to the Free Software Foundation, Inc.,
51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
```

We should also add a LICENSE.md to the repo.

**Comment by pboutros** [comment]
Yup, and the copyright should be OICR until 2018 and then UCLA from then on

**Comment by pboutros** [comment]
[Example of a line that should be changed](https://github.com/uclahs-cds/public-R-BoutrosLab-plotting-general/blob/b1ca77f9e2fa93f86c57d4f6af4205f721b31812/R/auto.axis.R#L1)

**Comment by forbiddenpersimmon** [comment]
When updating licenses, how should references to OICR [like here](https://github.com/uclahs-cds/package-BoutrosLab-plotting-general/blob/448df842de311386059b6285fa5e843c14306778/R/auto.axis.R#L4), be done? Should it be a combination of OICR/UCLA or just UCLA?

**Comment by dan-knight** [comment]
> Should it be a combination of OICR/UCLA or just UCLA?

I'm not sure! I wonder if it depends on the specific language. Some of these lines seem to reference ongoing responsibility, so I'd assume that these would be UCLA only. However, some also reference the past, like "credit to OICR scientists."

@aholmes Do you happen to have any knowledge on this?

**Comment by aholmes** [comment]
@pboutros do you have a prior agreement with OICR allowing us to relicense software they hold the copyright for? Otherwise, the licenses applied require that they remain in place:

GPLv1

> 1. You may copy and distribute verbatim copies of the Program's source
code as you receive it, in any medium, provided that you conspicuously and
appropriately publish on each copy an appropriate copyright notice and
disclaimer of warranty; **keep intact all the notices that refer to this
General Public License** ...

Artistic License 2.0

> (2) You may Distribute verbatim copies of the Source form of the Standard Version of this Package in any medium without restriction, either gratis or for a Distributor Fee, **provided that you duplicate all of the original copyright notices** ...

**Comment by dan-knight** [comment]
Here's a combined OICR and JCCC version. A few questions:

- Should we use JCCC or UCLA as the abbreviation?
- Am I correct to assume that "UCLA Jonsson Comprehensive Cancer Center" would be the correct institution rather than "University of California, Los Angeles"?
- I included both institutions in every mention of OICR.

```
The <package name> package is copyright (c) 2018 Ontario Institute for Cancer Research (OICR) and c() 2019 UCLA Jonsson Comprehensive Cancer Center (JCCC). This package and its accompanying libraries is free software; you can redistribute it and/or modify it under the terms of the GPL (either version 1, or at your option, any later version) or the Artistic License 2.0.  Refer to LICENSE for the full license text. OICR and JCCC make no representations whatsoever as to the SOFTWARE contained herein. It is experimental in nature and is provided WITHOUT

WARRANTY OF MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE OR ANY OTHER WARRANTY, EXPRESS OR IMPLIED. OICR MAKES NO REPRESENTATION
OR WARRANTY THAT THE USE OF THIS SOFTWARE WILL NOT INFRINGE ANY PATENT OR OTHER PROPRIETARY RIGHT.
By downloading this SOFTWARE, your Institution hereby indemnifies OICR and JCCC against any loss, claim, damage or liability, of whatsoever kind or
nature, which may arise from your Institution's respective use, handling or storage of the SOFTWARE. f publications result from research using this SOFTWARE, we ask that the Ontario Institute for Cancer Research and UCLA Jonsson Comprehensive Cancer Center be acknowledged and/or credit be given to OICR and JCCC scientists, as scientifically appropriate.

```

**Comment by pboutros** [comment]
Suggest you use UCLA as the parent institution, and date-ranges

________________________________

UCLA HEALTH SCIENCES IMPORTANT WARNING: This email (and any attachments) is only intended for the use of the person or entity to which it is addressed, and may contain information that is privileged and confidential. You, the recipient, are obligated to maintain it in a safe, secure and confidential manner. Unauthorized redisclosure or failure to maintain confidentiality may subject you to federal and state penalties. If you are not the intended recipient, please immediately notify us by return email, and delete this message from your computer.
